### ISSUE TITLE
bug: reading property of undefined with "no-reset-all-state-on-prop-change"

### ISSUE THREAD

**Issue by GeKorm** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#55]
### Did you check docs and existing issues?

- [x] I have read the plugin docs
- [x] I have searched the existing issues
- [x] I have read [You Might Not Need An Effect](https://react.dev/learn/you-might-not-need-an-effect) and am reasonably confident that I *do* need an effect

### Plugin version

0.8.1

### ESLint version

9.39.2

### Describe the bug

`getUseStateDecl` can return `undefined`, but there is unsafe access at
https://github.com/NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect/blob/3f33d604a299da6384fb24ac0d3ff1bb73dba991/src/rules/no-reset-all-state-on-prop-change.js#L89

### Steps To Reproduce

```tsx
import { useEffect, useState, useTransition } from 'react';

const Foo = () => {
  const [_0, setState] = useState(false);
  const [_1, startTransition] = useTransition();

  useEffect(() => {
    startTransition(() => {
      setState(true);
    });
  }, []);

  return null;
};
```

### Expected Behavior

No error

**Comment by NickvanDyke** [comment]
This should be fixed in v0.8.2 - let me know if not!
